src/tests/multihost/ipa/test_subid_ranges.py

```diff
@@ -0,0 +1,132 @@
+        uid_range = int(user_details[6].split(': ')[1].split('\n')[0])
+        gid_start = int(user_details[7].split(': ')[1].split('\n')[0])
+        gid_range = int(user_details[8].split(': ')[1].split('\n')[0])
+        ssh2 = SSHClient(multihost.client[0].sys_hostname,
+                         username=user, password=test_password)
+        cmd = execute_cmd(multihost, "ps -ef | grep bash").stdout_text
+        find_admin = f"ps -ef | grep bash | grep {user}"
+        if user in cmd:
+            proces_id = [int(i)
+                         for i in execute_cmd(multihost,
+                                              find_admin).stdout_text.split()
+                         if i.isdigit()][0]
+            multihost.client[0].run_command(f'kill -9 {proces_id}')
+        (results1, results2, results3) = ssh1.exec_command('unshare -U bash')
+        proces_id = [int(i)
+                     for i in execute_cmd(multihost,
+                                          find_admin).stdout_text.split()
+                     if i.isdigit()][0]
+        ssh1.exec_command(f"newuidmap {proces_id} "
+                          f"{uid_start} {uid_start + 1} 1")
+        ssh1.exec_command(f"newgidmap "
+                          f"{proces_id} {uid_start} {uid_start + 1} 1")
+        result = execute_cmd(multihost, f"cat /proc/{proces_id}/uid_map")
+        assert f'{uid_start}' in result.stdout_text
+        assert f'{uid_start + 1}' in result.stdout_text
+        result = execute_cmd(multihost, f"cat /proc/{proces_id}/gid_map")
+        assert f'{gid_start}' in result.stdout_text
+        assert f'{gid_start + 1}' in result.stdout_text
+        multihost.client[0].run_command(f'kill -9 {proces_id}')
+        ssh2.close()
+        ssh1.close()
+
+    def test_list_subid_ranges(self, multihost,
+                               create_bkp_for_subid_files):
+        """
+        :Title: support subid ranges managed by FreeIPA
+        :id: 4ab33f84-00c8-11ec-ad91-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1803943
+        :steps:
+            1. Configure subid: sss on /etc/nsswitch.conf
+            2. Test list_subid_ranges command
+            3. Test list_subid_ranges -g command
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+        """
+        multihost.client[0].run_command("echo 'subid: sss' "
+                                        ">> /etc/nsswitch.conf")
+        ssh1 = SSHClient(multihost.client[0].sys_hostname,
+                         username=user, password=test_password)
+        (result, result1, exit_status) = ssh1.execute_cmd('kinit',
+                                                          stdin=test_password)
+        assert exit_status == 0
+        (result, result1, exit_status) = ssh1.exec_command('klist')
+        assert user in str(result1.read())
+        (result, result1, exit_status) = ssh1.exec_command(f"ipa subid-find"
```

Comment:
Looking at this: `subid-generate` isn't executed here => you expect `test_subid_feature()` was executed first.
But how was it (`test_subid_feature()`) expected to work without 'nsswitch::conf' defined?
I'm confused.

Reply:
Yes . I will move subid-generate command  to class fixture "install_list_subid_ranges" .

Reply:
I would either rename `install_list_subid_ranges` (because it is also doing other things) or split it into few fixtures.

Reply:
renamed to prepare_list_subid_ranges

Reply:
Why is 'prepare_list_subid_ranges' a class fixture? It is only needed for `test_list_subid_ranges()`.
I don't say this is wrong, merely want to understand.

Reply:
And other way around: why `create_bkp_for_subid_files` is  function fixture and not a class fixture?

Reply:
"prepare_list_subid_ranges" is a class fixture as in the class fixture here i am putting only those lines which are require to run only once for entire test. These are commands need to run not only one test but all the test functions .

Reply:
> Why is 'prepare_list_subid_ranges' a class fixture? It is only needed for `test_list_subid_ranges()`.
> I don't say this is wrong, merely want to understand.

Ah, because it also does `subid-generate`. Ok, this needs to be class fixture.
Name is still not misleading, though.

Reply:
`prepare_list_subid_ranges` name is misleading as this function also setups subid ranges on IPA server besides installing a tool. I would suggest moving `subid-generate` and modification of `/etc/nsswitch.conf` to a third fixture (something like "environment setup")

Reply:
done